persistence/sql/src/test/java/org/infinispan/persistence/sql/AbstractSQLStoreFunctionalTest.java

```diff
@@ -0,0 +1,449 @@
+package org.infinispan.persistence.sql;
+
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertNull;
+
+import java.io.File;
+import java.lang.reflect.Method;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.concurrent.CompletionException;
+import java.util.function.Consumer;
+
+import javax.transaction.NotSupportedException;
+import javax.transaction.SystemException;
+import javax.transaction.TransactionManager;
+
+import org.infinispan.Cache;
+import org.infinispan.commons.CacheConfigurationException;
+import org.infinispan.commons.dataconversion.MediaType;
+import org.infinispan.commons.test.CommonsTestingUtil;
+import org.infinispan.commons.test.Exceptions;
+import org.infinispan.commons.util.Util;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.persistence.BaseStoreFunctionalTest;
+import org.infinispan.persistence.jdbc.common.DatabaseType;
+import org.infinispan.persistence.jdbc.common.UnitTestDatabaseManager;
+import org.infinispan.persistence.jdbc.common.configuration.ConnectionFactoryConfiguration;
+import org.infinispan.persistence.jdbc.common.configuration.ConnectionFactoryConfigurationBuilder;
+import org.infinispan.persistence.jdbc.common.connectionfactory.ConnectionFactory;
+import org.infinispan.persistence.sql.configuration.AbstactSchemaJdbcConfigurationBuilder;
+import org.infinispan.test.data.Address;
+import org.infinispan.test.data.Key;
+import org.infinispan.test.data.Person;
+import org.infinispan.test.data.Sex;
+import org.infinispan.transaction.TransactionMode;
+import org.postgresql.Driver;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+public abstract class AbstractSQLStoreFunctionalTest extends BaseStoreFunctionalTest {
```

Comment:
It would be good if we had a similar test to this in `server/tests`. That way QE can test against different DB containers easily.

\cc @gustavolira @diegolovison